R/runs.R

```diff
@@ -672,6 +678,49 @@ log_artifact <- function(x, FUN = saveRDS, filename, run_id, ...) {
   s3_file
 }
 
+#' @importFrom aws.s3 put_object
+#' @rdname log_artifact
+#' @export
+log_artifact.ggplot <- function(x, FUN, filename, run_id, ...) {
+
+  ## based on https://github.com/hrbrmstr/hrbrthemes/blob/master/R/aaa.r
+  if (isFALSE(requireNamespace("ggplot2", quietly = TRUE))) {
+    abort(
+      "Package `ggplot2` required for `ggsave`.\n",
+      "Please install and try again."
+    )
+  }
+
+  run_id <- resolve_run_id(maybe_missing(run_id))
+
+  experiment_id <- get_experiment_from_run(
+    run_id = run_id
+  )
+
+  s3_info <- get_s3_bucket_and_prefix()
+  s3_file <- create_s3_path(
+    s3_prefix = s3_info$prefix,
+    experiment_id = experiment_id,
+    run_id = run_id,
+    fname = filename
+  )
+
+  ext_pos <- regexpr("\\.([[:alnum:]]+)$", filename)
+  ext <- ifelse(ext_pos > -1L, substring(x, ext_pos + 1L), "")
```

Comment:
this is basically `tools::file_ext`

Reply:
i don't need this if i just choose to use `tempfile()` without specifying `fileext`, but i think it's probably best to specify `fileext`?